helix-core/src/main/java/org/apache/helix/HelixManager.java

```diff
@@ -419,6 +419,17 @@ void addExternalViewChangeListener(org.apache.helix.ExternalViewChangeListener l
    */
   Long getSessionStartTime();
 
+  /**
+   * Checks whether the cluster manager is leader and returns the session ID associated to the
+   * connection of cluster data store, if and only if it is leader.
+   *
+   * @return {@code String} session ID is a valid string if the cluster manager is leader.
+   * Otherwise, returns null.
+   */
+  default String getSessionIdIfLead() {
```

Comment:
If you think my previous suggestion about the SessionAwareHelixManager interface is too much, then please still keep the Optional<String> here.